core/DataAccess/ArchiveSelector.php

```diff
@@ -67,15 +68,28 @@ public static function getArchiveIdAndVisits(ArchiveProcessor\Parameters $params
         $doneFlags      = Rules::getDoneFlags($plugins, $segment);
         $doneFlagValues = Rules::getSelectableDoneFlagValues();
 
-        $results = self::getModel()->getArchiveIdAndVisits($numericTable, $idSite, $period, $dateStartIso, $dateEndIso, $minDatetimeIsoArchiveProcessedUTC, $doneFlags, $doneFlagValues);
-
+        $results = self::getModel()->getArchiveIdAndVisits($numericTable, $idSite, $period, $dateStartIso, $dateEndIso, null, $doneFlags, $doneFlagValues);
```

Comment:
not quite sure I understand why `minDatetimeIsoArchiveProcessedUTC ` is passed as `null` here? Wouldn't it make things faster to return immediately if no archive found that is newer than `minDatetime`? And in line `82` it might return even if the archive is older?

Reply:
This might have been to fix a test, but I can't remember anymore. I'll see if the build passes.